### PR TITLE
Implementation of Zero-Reference Deep Curve Estimation for unsupervised low-light enhancement

### PR DESCRIPTION
**Pull request by soumik12345**
Add implementation of [**Zero-Reference Deep Curve Estimation (Zero-DCE) for Low-Light Image Enhancement**](https://li-chongyi.github.io/Proj_Zero-DCE.html) as per https://github.com/soumik12345/restorers/issues/22
 - Primary Reference: https://github.com/Li-Chongyi/Zero-DCE
 - Secondary Reference 1: https://keras.io/examples/vision/zero_dce/
 - Secondary Reference 2: https://github.com/bsun0802/Zero-DCE
 - Secondary Reference 3: https://github.com/tuvovan/Zero_DCE_TF

**TODOs:**

- [x] Add Deep Curve Estimation Network
- [x] Add zero-reference loss functions
 - color constancy loss
 - exposure loss
 - illumination smoothness
- [x] Add Spatial Constancy Loss
- [x] Add training framework for training a Zero-reference model for low-light enhancement
- [x] Add corresponding unit tests
- [x] Add detailed documentation and references in the form of docstrings
- [x] Create an example demonstrating the usage of the Zero-DCE implementation

### CODE REVIEW

**Comment by soumik12345** [comment]
@SauravMaheshkar A small help is needed. Can you please check why only 13 unit tests are running even thought there are 16 unit tests in the source code?

Reference: https://github.com/soumik12345/restorers/actions/runs/3971016969/jobs/6807423368

**Comment by SauravMaheshkar** [comment]
@soumik12345 I addressed your comment in [6f3c268](https://github.com/soumik12345/restorers/pull/23/commits/6f3c26884bc517ff0cb712794057e3daecb58bba), as you can see in the [logs](https://github.com/soumik12345/restorers/actions/runs/3971526534/jobs/6808518293#step:4:1258) 16 tests are discovered now. 

The issue was because of the file name. `unittest` is not that smart with discovering file paths, it needs every file to be named `test_*.py`

File: restorers/model/zero_dce/zero_dce_model.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        x = x + r6 * (tf.square(x) - x)
+        x = x + r7 * (tf.square(x) - x)
+        enhanced_image = x + r8 * (tf.square(x) - x)
+        return enhanced_image
+
+    def call(self, data):
+        dce_net_output = self.dce_net(data)
+        return self.get_enhanced_image(data, dce_net_output)
+
+    def compute_losses(self, data, output):
```
Comment (SauravMaheshkar) [inline restorers/model/zero_dce/zero_dce_model.py:53]: I love the `compute_losses` fn here ❤️ . It cleans up the train step a lot.

**Comment by soumik12345** [comment]
Hi @SauravMaheshkar @ariG23498 
The PR is ready for review. After you are satisfied with the code and API design, I will add detailed docstrings.

File: examples/unsupervised_low_light_enhancement/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
+"""
+CLI Usage:
+train.py:
```
Comment (SauravMaheshkar) [inline examples/unsupervised_low_light_enhancement/evaluate.py:3]: This should be updated to `evaluate.py`

**Review by SauravMaheshkar — APPROVED** [review]
LGTM ❤️ !

As said I love the `compute_losses` function, really cleans up the `train_step`

**Comment by soumik12345** [comment]
Thanks a lot for your review @SauravMaheshkar.
I'll proceed with the docstrings.

**Review by SauravMaheshkar — APPROVED** [review]
LGTM ❤️

**Comment by soumik12345** [comment]
Thanks a lot for your reviews and feedback @SauravMaheshkar
